src/rules-engine/ThingConnection.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.*
+ */
+
+const e2p = require('event-to-promise');
+const Settings = require('../models/settings');
+const WebSocket = require('ws');
+
+class ThingConnection {
```

Comment:
Please add a short description of what a ThingConnection does for documentation.